sdk/program/src/loader_upgradeable_instruction.rs

```diff
@@ -0,0 +1,93 @@
+//! Instruction types
+
+use crate::{
+    pubkey::Pubkey,
+};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum UpgradeableLoaderInstruction {
```

Comment:
Let's extend `LoaderInstruction` instead.   Loaders that aren't upgradable don't need to implement the new instructions tacked on the end

Reply:
Not following, why would no-upgradeable loaders implemnt `UpgradeableLoaderInstruction`?  Also, what did you have in mind for `extend`?